### ISSUE TITLE
How can we get MSII merged from all samples mgf file output (like xcms) and can also related to precursor in asari ?

### ISSUE THREAD

**Issue by YUANMENG-1** [shuzhao-li-lab/asari#93]
How can we get MSII merged from all samples mgf file output (like xcms) and can also related to precursor in asari ?

**Comment by jmmitc06** [comment]
Currently the Asari feature table is only for ions detected in MS1 only. 

The issue of outputting the MS2 data in mgf or other format is in the to-do list for future versions of Asari or another tool in the Asari ecosystem. In the meantime, you can work with DDA data using the pcpfm pipeline based on Asari: https://github.com/shuzhao-li-lab/PythonCentricPipelineForMetabolomics.

**Comment by YUANMENG-1** [comment]
I've learned the complete pipeline you mentioned several times before. However, when directly processing the input mzML files, the output includes annotated results, but it doesn't have a merged MGF output like XCMS does.

For instance, if I want to use the first-level quantification table generated by Asari and annotate it with another pipeline, it cannot be done (because Asari or PCPFM cannot output second-level MGF files)?

**Comment by jmmitc06** [comment]
True, neither tool outputs the MS2 data in mgf format. For the time being
though, the empirical compounds generated by PCPFM will have the MS2
spectra associated with them in the JSON files. From those data, it should
be possible to convert them to mgf. However, any MS2 spectra that could not
be mapped to a feature will be excluded.

Ill have to think about how best to implement this and where. For instance,
should it be in Asari or the pipeline and how should mutliple MS2 spectra
be combined when they represent the same ion.

**Comment by jmmitc06** [comment]
v1.14 will have a very rudimentry version of this. Currently I'm dumping the pymzml representation of all ms2 spectra along with the other pickles. From that, it should be easy to process the MS2 however, we, you or I chose. I can have them output the mgf as well, but that is a next step. I don't want to do the fitlering and merging since that goes beyond pre-processing IMO. 

I'll close this since this is 'solved' IMO but will need more improvement. Feel free to reopen if future versions give problems.
